Dockerfile-slim.template

```diff
@@ -3,8 +3,31 @@ FROM postgres:%%PG_MAJOR%%
 ENV POSTGIS_MAJOR %%POSTGIS_MAJOR%%
 ENV POSTGIS_VERSION %%POSTGIS_VERSION%%
 
-RUN \
-      apt-get update && apt-get install -y --no-install-recommends \
-              postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR=$POSTGIS_VERSION \
-              postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR-scripts \
-      && rm -rf /var/lib/apt/lists/*
+RUN set -ex \
+    # RDS usually lags behind the version of PostGIS maintained in the main
+    # repository. ca-certificates is required due to a 301 HTTP → HTTPS redirect
+    # from apt-archive.postgresql.org. apt-transport-https is required for
+    # stretch-based images (9.6, 10.9, and 11.4).
+    && apt-get update && apt-get install -y apt-transport-https ca-certificates --no-install-recommends \
+    && versionCodename=$(env -i bash -c ". /etc/os-release; echo \$VERSION | sed 's/.*(\(.*\))/\1/'") \
+    && echo "deb http://apt-archive.postgresql.org/pub/repos/apt/ $versionCodename-pgdg-archive main" > /etc/apt/sources.list.d/pgdg-archive.list \
```

Comment:
I came across some anecdotal concerns about using `/etc/os-release` as an attack vector for injection when sourced. I tried my best to work around this by running the command in a new shell with an isolated environment (`env -i`).

I had to use pattern matching to parse the codename from a larger variable because some older variants of Debian (still used by upstream PostgreSQL images) don't contain [`$VERSION_CODENAME`](https://manpages.debian.org/unstable/systemd/os-release.5.en.html).

I'm not super happy because:
1. I've never used `env -i` before, and I'm not sure what the potential is for someone to break out of that environment.
2. The pattern matching isn't readable at all.
3. There wasn't any prior art on this. I don't know if this is idiomatic. It makes me feel uncomfortable to be designing my own thing for something that feels like it should be clear-cut.